crypto/src/signature.rs

```diff
@@ -134,6 +139,33 @@ pub(crate) fn verify_bls_sig(data: &[u8], pub_k: &[u8], sig: &Signature) -> bool
     verify(&sig, &[hashed], &[pk])
 }
 
+pub(crate) fn verify_agg_bls_sig(data: &[&[u8]], pub_keys : &[&[u8]], aggregate_sig: &BlsSignature) -> bool{
+
+    // If the number of public keys and data does not match, then return false
+    if data.len()  != pub_keys.len() {
+        return false;
+    }
+
+    let num_sigs = data.len();
+    let mut pks : Vec<BlsPubKey> = vec!();
+    let mut hashed_data : Vec<G2>  = vec!();
+
+    // For loop produces the hashed data and gets the public keys from bytes.
+    for x in 0 .. num_sigs{
+        let pk = match BlsPubKey::from_bytes(pub_keys[x]) {
+            Ok(v) => v,
+            Err(_) => return false,
+        };
+        pks.push(pk);
+        let h_data =   bls_hash(data[x]);
+        hashed_data.push(h_data);
+    }
```

Comment:
These two vectors do not need to be handled in the same function, this can be made more easy to read and more efficient if you just map the values to the vector types expected (can explain in person if this is confusing)